src/Razor/src/Microsoft.VisualStudio.Editor.Razor/TelemetryReporter.cs

```diff
@@ -247,6 +254,26 @@ public IDisposable TrackLspRequest(string lspMethodName, string languageServerNa
         }));
     }
 
+    private static bool IsNumeric(object? o)
+        => o is not null &&
+        !o.GetType().IsEnum &&
+        Type.GetTypeCode(o.GetType()) switch
+        {
+            TypeCode.Char or
+            TypeCode.SByte or
+            TypeCode.Byte or
+            TypeCode.Int16 or
+            TypeCode.Int32 or
+            TypeCode.Int64 or
+            TypeCode.Double or
+            TypeCode.Single or
+            TypeCode.UInt16 or
+            TypeCode.UInt32 or
+            TypeCode.UInt64
+            => true,
+            _ => false
+        };
```

Comment:
Consider removing the switch expression, since it's just a boolean expression he returned. Also, should `System.Decimal` allowed?

```suggestion
        => o?.GetType() is Type type &&
           !type.IsEnum &&
           Type.GetTypeCode(type) is (
               TypeCode.Char or
               TypeCode.SByte or
               TypeCode.Byte or
               TypeCode.Int16 or
               TypeCode.Int32 or
               TypeCode.Int64 or
               TypeCode.Double or
               TypeCode.Single or
               TypeCode.UInt16 or
               TypeCode.UInt32 or
               TypeCode.UInt64);
```